File: src/main/java/ir/mmd/intellijDev/Actionable/caret/editing/settings/UI.java
package ir.mmd.intellijDev.Actionable.caret.editing.settings;

import javax.swing.*;
import java.awt.event.MouseAdapter;
import java.awt.event.MouseEvent;

import static ir.mmd.intellijDev.Actionable.caret.editing.settings.SettingsState.Defaults.showPasteActionHints;

public class UI {
	private JPanel component;
	private JCheckBox showPasteActionHintsCheckbox;
	private JButton showPasteActionHintsCheckboxDefaultButton;
	
	public UI() { initListeners(); }
	
	private void initListeners() {
		showPasteActionHintsCheckboxDefaultButton.addMouseListener(new MouseAdapter() {
			@Override
			public void mouseClicked(MouseEvent e) {
				showPasteActionHintsCheckbox.setSelected(showPasteActionHints);
			}
		});
	}
	
	public JPanel getComponent() { return component; }
	
	public boolean isPasteActionHintsShown() { return showPasteActionHintsCheckbox.isSelected(); }
	public void setPasteActionHintsShown(boolean b) { showPasteActionHintsCheckbox.setSelected(b); }
}

File: src/main/java/ir/mmd/intellijDev/Actionable/caret/editing/settings/UI.kt
package ir.mmd.intellijDev.Actionable.caret.editing.settings

import com.intellij.ui.dsl.builder.*
import ir.mmd.intellijDev.Actionable.ActionableBundle
import ir.mmd.intellijDev.Actionable.util.observableMutablePropertyOf

class UI {
	private val isPasteActionHintsShownProperty = observableMutablePropertyOf(SettingsState.Defaults.showPasteActionHints)
	var isPasteActionHintsShown by isPasteActionHintsShownProperty
	
	val component = panel {
		row {
			checkBox(ActionableBundle.string("caretEditingPanel.showPasteActionHintsCheckBox.label"))
				.bindSelected(isPasteActionHintsShownProperty)
				.align(Align.FILL)
				.comment(ActionableBundle.string("caretEditingPanel.showPasteActionHintsCheckBox.comment"))
			
			button(ActionableBundle.string("global.defaultButton.label")) {
				isPasteActionHintsShown = SettingsState.Defaults.showPasteActionHints
			}.align(AlignX.RIGHT + AlignY.CENTER)
		}
	}
}
